### PR TITLE
[WIP] feat: oxc_transforms_jsx

### PR DESCRIPTION
**Pull request by magic-akari**
I am working on this (slowly)

### CODE REVIEW

**Comment by Boshen** [comment]
OMG this is amazing!

File: crates/oxc_transforms_jsx/src/automatic.rs [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+};
+use oxc_ast::Atom;
+use oxc_ast::Span;
+use oxc_ast::{
+    ast::{Expression, JSXElement},
+    visit_mut::VisitMut,
+    AstBuilder,
+};
+
+pub struct Automatic<'a> {
```
[the remark this answers is not in the document]
Comment (magic-akari) [reply crates/oxc_transforms_jsx/src/automatic.rs:15]: It's the term from Babel.
https://babeljs.io/docs/babel-preset-react#runtime

There are two runtimes: `classic` and `automatic`

File: crates/oxc_transforms_jsx/src/automatic.rs [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+}
+
+impl<'a> Automatic<'a> {
+    pub fn new(
+        allocator: &'a Allocator,
+        import_source: Option<Atom>,
+        development: Option<bool>,
+    ) -> Self {
+        Self {
+            import_source: import_source.unwrap_or_else(|| Atom::from("React")),
```
Comment (thepassle) [inline crates/oxc_transforms_jsx/src/automatic.rs:28]: Should react be hardcoded here? What if some uses Preact?

**Comment by Boshen** [comment]
How complicated is JSX transform? I'm considering to do this in the AST -> HIR process.
Is there a open spec for the transformation part?

**Comment by Boshen** [comment]
This seems easy https://github.com/babel/babel/blob/main/packages/babel-plugin-transform-react-jsx/src/create-plugin.ts

I'm really keen to implement this in the ast -> hir lowering step here 

https://github.com/Boshen/oxc/blob/7148cb8c008e97581342b0431068d68cab9fff84/crates/oxc_ast_lower/src/lib.rs#L520-L566

In this way every we remove a level of indirection in the HIR

So the only hard decision is on the runtime ...

**Comment by Boshen** [comment]
Let's continue the work inside the ast -> hir lowering process once it's ready. Thank you for taking the time on this, I'll base the work on top of your work.

**Pull request closed without merging**